### DISCUSSION TITLE
Ecs separation

### DISCUSSION THREAD

**Discussion by fzfking** [friflo/Friflo.Json.Fliox#36] [Ideas]
Maybe make separate repos to ecs and orm? It will simplify work with documentation

**Comment by friflo** [comment]
Hi,

at the moment I cannot move to a new project.
Simplify the GitHub star count will start at 0 and the ECS will never get of the ground :)

Maybe at some point in the future. But don't know when.

**Reply by sergiyha** [reply]
If you want to get ECS of the ground, try releasing it on the Unity asset store. There are a lot of ECS enthusiasts)

**Reply by friflo** [reply]
Do have a link to an ECS in the Unity asset store which is a good example?

**Reply by sergiyha** [reply]
https://assetstore.unity.com/packages/tools/game-toolkits/entitas-87638

**Reply by friflo** [reply]
btw: I decided to create a Unity specific extension for the Unity Asset Store.

I also plan to add an HTTP/WebSocket server to enable browsing and access via REST API.
The Web UI will look like: http://ec2-18-215-176-108.compute-1.amazonaws.com/fliox/

**Reply by sergiyha** [reply]
Will ecs engine be separated especially for unity and released on asset store? 

I am not quite shure about second statement. Do you want to make debug runtime entity's monitoring web ui?

**Reply by friflo** [reply]
yes, a separate repo.
It will contain Friflo.Engine.ECS.dll as binary.
I will try to release it on the Asset Store.

I will integrate HTTP server functionality - optional.
You can then use a Browser to explore the ECS database.
It is useful for debugging I think.
It also enables to query or update the ECS database with external tools via HTTP / JSON
like JS or Python.

It will also enable connecting multiple game instances on a local network.
One instance is a server - the others clients. The ECS state will be synchronized between game instances.

**Reply by sergiyha** [reply]
Sounds great!

**Reply by friflo** [reply]
I think this Unity extension is almost finished. Will publish it in 1-4 weeks.
Just added two screenshots to https://github.com/friflo/Friflo.Json.Fliox/wiki/Extensions

**Reply by friflo** [reply]
finally moved to ECS to its own repo: https://github.com/friflo/Friflo.Engine.ECS

**Comment by friflo** [comment]
Regarding documentation. Not sure if its an alternative for you.
You may use the [API Reference](https://github.com/friflo/Friflo.Engine-docs/blob/main/README.md)

**Comment by fzfking** [comment]
Okey, understand

**Reply by friflo** [reply]
Want to leave this discussion open.
Had already the same request on reddit.
